src/devices/src/virtio/vsock/device.rs

```diff
@@ -65,6 +69,66 @@ pub struct Vsock<B> {
     pub(crate) device_state: DeviceState,
 }
 
+#[derive(Clone, Versionize)]
+pub struct VsockState {
```

Comment:
I think this is a good potential example why keeping state as constituent part of a device post-restore might often not be a good fit. In this particular case, as soon as the `QueueState` type is no longer identical to `Queue` itself, the vsock device ends up holding both a `Vec` of `Queue`s and one of `QueueState`s if it contains a `VsockState` member during normal operation.